app/src/main/java/io/github/droidkaigi/confsched2017/view/activity/BaseActivity.java

```diff
@@ -3,31 +3,40 @@
 import com.tomoima.debot.Debot;
 
 import android.content.Context;
+import android.net.Uri;
 import android.os.Bundle;
 import android.support.annotation.IdRes;
 import android.support.annotation.LayoutRes;
 import android.support.annotation.NonNull;
+import android.support.customtabs.CustomTabsIntent;
 import android.support.v4.app.Fragment;
 import android.support.v4.app.FragmentTransaction;
+import android.support.v4.content.ContextCompat;
 import android.support.v7.app.ActionBar;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.app.AppCompatDelegate;
 import android.support.v7.widget.Toolbar;
+import android.text.TextUtils;
 import android.view.KeyEvent;
 import android.view.MenuItem;
+import android.webkit.URLUtil;
 
 import io.github.droidkaigi.confsched2017.MainApplication;
+import io.github.droidkaigi.confsched2017.R;
 import io.github.droidkaigi.confsched2017.di.ActivityComponent;
 import io.github.droidkaigi.confsched2017.di.ActivityModule;
+import io.github.droidkaigi.confsched2017.model.Session;
+import io.github.droidkaigi.confsched2017.view.helper.Navigator;
 import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper;
 
-public abstract class BaseActivity extends AppCompatActivity {
+public abstract class BaseActivity extends AppCompatActivity implements Navigator {
```

Comment:
It looks like we've crisscrossed our comments.

I'm going to create non-final `Navigator` class that holding activity since interface is no longer needed.